src/setup_payload/AdditionalDataPayloadGenerator.cpp

```diff
@@ -51,35 +51,34 @@ AdditionalDataPayloadGenerator::generateAdditionalDataPayload(uint16_t lifetimeC
 {
     System::PacketBufferTLVWriter writer;
     TLVWriter innerWriter;
-    char rotatingDeviceIdBuffer[RotatingDeviceId::kHexMaxLength];
+    uint8_t rotatingDeviceIdBuffer[RotatingDeviceId::kHexMaxLength];
     size_t rotatingDeviceIdBufferSize = 0;
 
     // Initialize TLVWriter
     writer.Init(chip::System::PacketBufferHandle::New(chip::System::PacketBuffer::kMaxSize));
 
-    ReturnErrorOnFailure(writer.OpenContainer(AnonymousTag, kTLVType_Structure, innerWriter));
+    ReturnErrorOnFailure(writer.OpenContainer(RotatingDeviceId::DiscoveryExtensionDescriptorTag, kTLVType_Structure, innerWriter));
```

Comment:
There is work agreed by @chrisdecenzo  and @bzbarsky-apple  to use Anonymous struct here: https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/3945/files#diff-ede0c618c5ffe4b09e528ceb9a9127a5abbff027e634d575c57eb7ca792853cdR214

I think the outer container should remain anonymous struct.